packages/@aws-cdk/aws-logs/lib/data-protection-policy.ts

```diff
@@ -0,0 +1,180 @@
+    this.description = props.description || 'cdk generated data protection policy';
+    this.version = '2021-06-01';
+
+    var findingsDestination: FindingsDestination = {};
+
+    if (props.cloudWatchLogsAuditDestination) {
+      findingsDestination.cloudWatchLogs = {
+        logGroup: props.cloudWatchLogsAuditDestination,
+      };
+    }
+
+    if (props.s3AuditDestination) {
+      findingsDestination.s3 = {
+        bucket: props.s3AuditDestination,
+      };
+    }
+
+    if (props.firehoseAuditDestination) {
+      findingsDestination.firehose = {
+        deliveryStream: props.firehoseAuditDestination,
+      };
+    }
+
+    var identifierArns: string[] = [];
+    for (let identifier of props.identifiers) {
+      identifierArns.push('arn:aws:dataprotection::aws:data-identifier/' + identifier);
+    };
+
+    var auditStatement: AuditStatement = {
+      sid: 'audit-statement-cdk',
+      dataIdentifier: identifierArns,
+      operation: {
+        audit: {
+          findingsDestination: findingsDestination,
+        },
+      },
+    };
+
+    var redactStatement: RedactStatement = {
+      sid: 'redact-statement-cdk',
+      dataIdentifier: identifierArns,
+      operation: {
+        deidentify: {
+          maskConfig: {},
+        },
+      },
+    };
+
+    this.statement = [auditStatement, redactStatement];
+  }
+
+}
+
+type AuditStatement = {
+  sid: string;
+  dataIdentifier: string[];
+  operation: AuditOperation;
+}
+
+type AuditOperation = {
+  audit: FindingsDestinations;
+}
+
+type FindingsDestinations = {
+  findingsDestination: FindingsDestination;
+}
+
+type FindingsDestination = {
+  cloudWatchLogs?: CloudWatchLogsDestination;
+  firehose?: FirehoseDestination;
+  s3?: S3Destination;
+}
+
+type CloudWatchLogsDestination = {
+  logGroup: string;
+}
+
+type FirehoseDestination = {
+  deliveryStream: string;
+}
+
+type S3Destination = {
+  bucket: string;
+}
+
+type RedactStatement = {
+  sid: string;
+  dataIdentifier: string[];
+  operation: DeidentifyOperation;
+}
+
+type DeidentifyOperation = {
+  deidentify: MaskConfig;
+}
+
+type MaskConfig = {
+  maskConfig: {}
+}
+
+
+/**
+ * Interface for creating a data protection policy
+ */
+export interface DataProtectionPolicyProps {
+  /**
+   * Name of the data protection policy
+   *
+   * @default - 'data-protection-policy-cdk'
+   */
+  readonly name?: string;
+
+  /**
+   * Description of the data protection policy
+   *
+   * @default - 'cdk generated data protection policy'
+   */
+  readonly description?: string;
+
+  /**
+   * List of data protection identifiers. Must be in the following list: https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/protect-sensitive-log-data-types.html
+   */
+  readonly identifiers: string[];
+
+  /**
+   * CloudWatch Logs log group to send audit findings to. The log group must already exist.
+   *
+   * @default - no CloudWatch Logs audit destination
+   */
+  readonly cloudWatchLogsAuditDestination?: string;
```

Comment:
This string would be an ARN and I don't think this is the right user experience here. This should probably be an ILogGroup.

Reply:
Also, can you clarify a point here? Can you specify all three here or are they mutually exclusive? If you can specify all three, can you specify multiple of any of the three or just one of each?

Reply:
This would be log group name, S3 bucket name, and delivery stream name respectively. You can specify none or up to all 3 types, but only one of each.

Reply:
I explored using ILogGroup, IBucket, and IDeliveryStream, but I couldn't use IDeliveryStream, due to what I believe is a circular dependency. Given that these are names and `logs` is unable to take a dependency on `kinesisfirehose`, to provide a consistent experience I left these fields as strings. The example in the README should clarify how these are used.